test/fs.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+const metatests = require('metatests');
+const fs = require('fs');
+const { mkdirp } = require('..');
+const path = require('path');
+const testMkdirp = metatests.test('mkdir');
+const mkdirpTestDir = 'test/ex1/ex2';
+
+testMkdirp.beforeEach((test, cb) => {
+  fs.rmdir(mkdirpTestDir, err => {
+    if (err && err.code === 'ENOENT') {
+      cb();
+      return;
+    }
+    test.error(err);
+    fs.rmdir(path.dirname(mkdirpTestDir), err => {
+      if (err && err.code !== 'ENOENT') {
+        test.error(err);
+      }
+      cb();
+    });
+  });
+});
+
+testMkdirp.afterEach((test, cb) => {
+  fs.rmdir(mkdirpTestDir, err => {
+    if (err) test.error(err);
+    cb();
+    fs.rmdir(path.dirname(mkdirpTestDir), err => {
+      if (err) test.error(err);
+    });
+    cb();
+  });
+});
+
+testMkdirp.test('create 2 directories using mode', test => {
+  mkdirp(mkdirpTestDir, 0o777, err => {
+    if (err) test.bailout(err, 'Cannot create directory');
```

Comment:
I think `test.error` should be enough.
```suggestion
    test.error(err, 'Cannot create directory');
```